pimd/pim_nht.c

```diff
@@ -760,33 +753,41 @@ int pim_parse_nexthop_update(ZAPI_CALLBACK_ARGS)
 					pnc->rpf.rpf_addr.u.prefix6;
 #endif
 				break;
-			case NEXTHOP_TYPE_IPV6_IFINDEX:
+#if PIM_IPV == 4
+			/* RFC5549 IPv4-over-IPv6 nexthop handling:
+			 * if we get an IPv6 nexthop in IPv4 PIM, hunt down a
+			 * PIM neighbor and use that instead.
+			 */
+			case NEXTHOP_TYPE_IPV6_IFINDEX: {
+				struct interface *ifp1 = NULL;
+				struct pim_neighbor *nbr = NULL;
+
 				ifp1 = if_lookup_by_index(nexthop->ifindex,
 							  pim->vrf->vrf_id);
 
 				if (!ifp1)
 					nbr = NULL;
 				else
+					/* FIXME: should really use nbr's
+					 * secondary address list here
+					 */
 					nbr = pim_neighbor_find_if(ifp1);
+
 				/* Overwrite with Nbr address as NH addr */
 				if (nbr)
-#if PIM_IPV == 4
 					nexthop->gate.ipv4 = nbr->source_addr;
-#else
-					nexthop->gate.ipv6 = nbr->source_addr;
-#endif
-				else {
+				else
 					// Mark nexthop address to 0 until PIM
 					// Nbr is resolved.
-#if PIM_IPV == 4
 					nexthop->gate.ipv4 = PIMADDR_ANY;
-#else
-					nexthop->gate.ipv6 = PIMADDR_ANY;
-#endif
-				}
 
 				break;
 			}
+#endif
+			default:
```

Comment:
errgg no.  please allow the failure to enumerate all the different cases to be caught by the compiler.  I am terribly tired of having to fix bugs because of default usage.

Reply:
done